src/app/services/User.service.ts

```diff
@@ -40,4 +41,17 @@ export class UserService {
     const savedUser = await this.repository.save(userToSave);
     return this.userMap.toDTO(savedUser);
   }
+
+  async updateSourcesOfInterest(token: string, sourcesOfInterest: Source[]) {
+    const claims = this.encryptionService.verify<UserTokenClaims>(token);
```

Comment:
This line is not needed. In the isAuthenticated.middleware the jwt is validated and if validation passed the user claims are attached to the request object. User id can be extracted from the request.user in the controller and passed to the UserService.updateSourcesOfInterest as an argument